src/Wolf.cpp

```diff
@@ -0,0 +1,466 @@
+}
+
+
+//calculate self term for a box
+double Wolf::BoxSelf(uint box) const
+{
+    if (box >= BOXES_WITH_U_NB){
+        return 0.0;
+    } else {
+        GOMC_EVENT_START(1, GomcProfileEvent::SELF_BOX);
+        double self = 0.0;
+        double molSelfEnergy;
+        uint i, j, length, molNum;
+        double lambdaCoef = 1.0;
+
+// Each thread calculates a kind
+#ifdef _OPENMP
+    #pragma omp parallel for default(none) private(molSelfEnergy, i, j, length, molNum, lambdaCoef) \
+    shared(box, mols, molLookup, lambdaRef, molSelfEnergies) \
+    reduction(+:self)
+#endif
+        for (i = 0; i < mols.GetKindsCount(); i++) {
+          MoleculeKind const& thisKind = mols.kinds[i];
+          molNum = molLookup.NumKindInBox(i, box);
+          molSelfEnergy = 0.0;
+          if(lambdaRef.KindIsFractional(i, box)) {
+              //If a molecule is fractional, we subtract the fractional molecule and
+              // add it later
+              --molNum;
+              //returns lambda and not sqrt(lambda)
+              lambdaCoef = lambdaRef.GetLambdaCoulomb(i, box);
+          }
+          // Store this for quick access in ChangeSelf
+          molSelfEnergy = molSelfEnergies[i];
+          self += (molSelfEnergy * molNum);
+          if(lambdaRef.KindIsFractional(i, box)) {
+              //Add the fractional molecule part
+              self += (molSelfEnergy * lambdaCoef);
+          }
+        }
+        // M_2_SQRTPI is 2/sqrt(PI), so need to multiply by 0.5 to get sqrt(PI)
+        // Vlugt
+        //self *= ((ff.wolfAlpha[box] * M_2_SQRTPI * 0.5) + ff.wolfFactor1[box]);
+        // we eliminate the alpha/root(pi) using Wolf,mod from Gross et al
+        // We eliminate the dampened rij term by choosing psi = 1
+        self *= -1.0 * ff.wolfFactor1[box] * num::qqFact;
+
+        GOMC_EVENT_STOP(1, GomcProfileEvent::SELF_BOX);
+        return self;
+    }
+}
+
+
+//calculate correction term for a molecule
+double Wolf::MolCorrection(uint molIndex, uint box) const
+{
+  if (box >= BOXES_WITH_U_NB)
+    return 0.0;
+
+  GOMC_EVENT_START(1, GomcProfileEvent::CORR_MOL);
+  double dist, distSq;
+  double correction = 0.0, dampenedCorr = 0.0, undampenedCorr = 0.0;
+  XYZ virComponents;
+
+  MoleculeKind& thisKind = mols.kinds[mols.kIndex[molIndex]];
+  uint atomSize = thisKind.NumAtoms();
+  uint start = mols.MolStart(molIndex);
+  double lambdaCoef = GetLambdaCoef(molIndex, box);
+
+  for (uint i = 0; i < atomSize; i++) {
+    if(particleHasNoCharge[start + i]) {
+      continue;
+    }
+    // This term only needs to be calculated once, if the molecules are rigid
+    // Or of the maximum radius of gyration < RCutCoulomb
+    // Otherwise, parts of the molecule may extend out of range of each other
+    // For now assume the latter.
+    for (uint j = i + 1; j < atomSize; j++) {
+      // Need to check for cutoff
+      if(currentAxes.InRcut(distSq, virComponents, currentCoords,
+                         start + i, start + j, box)){
```

Comment:
The rcut that boxDim uses to check the distance is max(rcut, rcutcoulomb). You need to additionally check for rcutcoulomb, otherwise if user set a cutoff that `rcut > rcutcoulomb` the results will be wrong.

Reply:
This same bug applies to CalcAdd1_4 in all the FFParticle derived classes.

Reply:
Your branch seems to be not up-to-date. Look at this: https://github.com/GOMC-WSU/GOMC/blob/development/src/FFParticle.cpp#L265:#L284